asmcomp/selectgen.ml

```diff
@@ -355,8 +356,8 @@ method effects_of exp =
   | Cop (op, args, _) ->
     let from_op =
       match op with
-      | Capply _ | Cextcall _ -> EC.arbitrary
-      | Cpoll | Calloc -> EC.none
+      | Capply _ | Cextcall _ | Cpoll | Cnop -> EC.arbitrary
+      | Calloc -> EC.none
```

Comment:
Is the change for `Cpoll` from `EC.none` to `EC.arbitrary` intentional here?

Reply:
The compiler had an assertion failure for `Cnop` having no effect since it doesn't return anything useful; such a primitive is not useful as far as the compiler can tell :-). I don't remember seeing the error with `Cpoll`. So it should probably be put back to what it way it was.

Reply:
Fixed.